scripts/weekly-summary

```diff
@@ -0,0 +1,374 @@
+            ess = "s" if (len(failed[name]) > 1) else ""
+            print(f"{name} failed {len(failed[name])} time{ess}", file=buf)
+            print("-" * 40, file=buf)
+            for f in failed[name]:
+                if "start_time" not in f:
+                    start_time = ""
+                else:
+                    start_time = datetime.fromtimestamp(f["start_time"]).strftime("%m/%d/%Y %H:%M:%S")
+
+                if "elapsed_time" not in f:
+                    elapsed_time = 0
+                else:
+                    elapsed_time = f["elapsed_time"]
+
+                # Get the result message
+                msg = f["result"].rsplit("FAILED:")[-1]
+                print(f'\n{start_time} ({elapsed_time}s): {msg}', file=buf)
+            print("\n", file=buf)
+
+        if len(failed):
+            print("\n", file=buf)
+
+    return buf.getvalue()
+
+
+def main(args, token):
+    artifacts = get_artifacts(token, args.start, args.end)
+    if args.verbose:
+        print(json.dumps(artifacts))
+    zipfiles = download_artifacts(token, artifacts)
+
+    for name, datename, f in zipfiles:
+        if args.rebuild or not os.path.exists(datename+".json"):
+            logs = extract_logs(f)
+
+            # This is needed for logs without timestamps
+            if args.rebuild:
+                rebuild_logs(logs)
+
+            # Run summary on kstest.log
+            cmd = ["log2json",
+                   "--group", name,
+                   "--output", datename+".json",
+                   os.path.join(logs.name, "kstest.log")
+            ]
+            if args.debug:
+                print(cmd)
+            check_call(cmd)
+
+    report = summary(args, (d+".json" for _, d, _ in zipfiles))
+    if args.output:
+        with open(args.output, "w") as f:
+            f.write(report)
+    else:
+        print(report)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="Generate a weekly summary of test results")
+    parser.add_argument("--start", type=datetime.fromisoformat,
+            default=(datetime.now() - timedelta(days=7)),
+            help="Start time. eg. 2022-03-03T03:46:11 (default is -7 days)")
+    parser.add_argument("--end", type=datetime.fromisoformat,
+            default=datetime.now(),
+            help="end time. eg. 2022-03-03T03:46:11 (default is now)")
+    parser.add_argument("--rebuild",
+            default=False, action="store_true",
+            help="Rebuild logs with timestamps")
+    parser.add_argument("--output",
+            help="Path and filename to write summary report to")
+    parser.add_argument("--debug", default=False, action="store_true")
+    parser.add_argument("--verbose", default=False, action="store_true")
+    args = parser.parse_args()
+
+    if "GITHUB_TOKEN" not in os.environ:
+        print("Set GITHUB_TOKEN environmental variable to github token with access to the artifact api.")
+        sys.exit(1)
```

Comment:
Future improvement would be to be able to pass this as argument too. I know this is great for GH actions but not really that nice for local execution.

Reply:
You really do not want to pass tokens on the cmdline since it is visible to other users. Yes, not a problem in this specific case, but generally not a good idea. config files and environmental variables are the best way to set these.